SU2_CFD/src/fluid/CSU2TCLib.cpp

```diff
@@ -1829,227 +1946,134 @@ void CSU2TCLib::DiffusionCoeffGY(){
+    /*--- Calculate species diffusion coefficient ---*/
+    DiffusionCoeff[iSpecies] = (gam_t*gam_t*Mi*(1-Mi*gam_i) / denom);
   }
-  // if (ionization) {
-  //TODO: Update correct iElectron....
-  //   iSpecies = nSpecies-1;
-
-  //   /*--- Initialize the species diffusion coefficient ---*/
-  //   DiffusionCoeff[iSpecies] = 0.0;
-
-  //   /*--- Calculate molar concentration ---*/
-  //   Mi      = MolarMass[iSpecies];
-  //   gam_i   = rhos[iSpecies] / (Density*Mi);
-  //   denom = 0.0;
-  //   for (jSpecies = 0; jSpecies < nHeavy; jSpecies++) {
-  //     if (iSpecies != jSpecies) {
-  //       Mj    = MolarMass[jSpecies];
-  //       gam_j = rhos[iSpecies] / (Density*Mj);
-
-  //       /*--- Calculate the Omega^(0,0)_ij collision cross section ---*/
-  //       Omega_ij = 1E-20 * Omega00(iSpecies,jSpecies,3)
-  //           * pow(Tve, Omega00(iSpecies,jSpecies,0)*log(Tve)*log(Tve)
-  //           + Omega00(iSpecies,jSpecies,1)*log(Tve)
-  //           + Omega00(iSpecies,jSpecies,2));
-
-  //       /*--- Calculate "delta1_ij" ---*/
-  //       d1_ij = 8.0/3.0 * sqrt((2.0*Mi*Mj) / (pi*Ru*Tve*(Mi+Mj))) * Omega_ij;
-
-  //       /*--- Calculate heavy-particle binary diffusion coefficient ---*/
-  //       D_ij = kb*Tve/(Pressure*d1_ij);
-  //       denom += gam_j/D_ij;
-  //     }
-  //   }
-  //   DiffusionCoeff[iSpecies] = gam_t*gam_t*MolarMass[iSpecies]*(1-MolarMass[iSpecies]*gam_i) / denom;
-  // }
 }
 
 void CSU2TCLib::ViscosityGY(){
 
-  su2double pi = PI_NUMBER;
-  su2double Na = AVOGAD_CONSTANT;
-  su2double Mu = 0.0;
+  const su2double Na = AVOGAD_CONSTANT;
+  Mu = 0.0;
 
   /*--- Mixture viscosity via Gupta-Yos approximation ---*/
-  for (iSpecies = 0; iSpecies < nHeavy; iSpecies++) {
+  for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
 
     su2double denom = 0.0;
 
     /*--- Calculate molar concentration ---*/
-    su2double Mi    = MolarMass[iSpecies];
-    su2double gam_i = rhos[iSpecies] / (Density*Mi);
+    const su2double Mi    = (MolarMass[iSpecies] + EPS);
+    const su2double gam_i = rhos[iSpecies] / (Density*Mi);
+
+    for (jSpecies = 0; jSpecies < nSpecies; jSpecies++) {
+      const su2double Mj    = (MolarMass[jSpecies] + EPS);
+      const su2double gam_j = rhos[jSpecies] / (Density*Mj);
 
-    for (jSpecies = 0; jSpecies < nHeavy; jSpecies++) {
-      su2double Mj    = MolarMass[jSpecies];
-      su2double gam_j = rhos[jSpecies] / (Density*Mj);
+      const su2double T_col = (iSpecies == 0 && ionization) ? Tve : T; 
 
-      /*--- Calculate "delta" quantities ---*/
-      su2double Omega_ij = 1E-20 * Omega11(iSpecies,jSpecies,3)
-          * pow(T, Omega11(iSpecies,jSpecies,0)*log(T)*log(T)
-          + Omega11(iSpecies,jSpecies,1)*log(T)
-          + Omega11(iSpecies,jSpecies,2));
-      su2double d2_ij = 16.0/5.0 * sqrt((2.0*Mi*Mj) / (pi*Ru*T*(Mi+Mj))) * Omega_ij;
+      su2double d2_ij = ComputeCollisionDelta(iSpecies, jSpecies, Mi, Mj, T_col, false);
+      if (d2_ij > 1E16) d2_ij = 1E16;
```

Comment:
Repeated, why don't you put the limit inside the function?
Or define a constant for the limit, otherwise this repetition is bound to cause problems at some point

Reply:
I moved it into ComputeCollisionDelta function